File: src/main/java/musiccube/entities/SongInstrument.java
package musiccube.entities;

import javax.persistence.*;

@Entity
public class SongInstrument{

    @Id
    @GeneratedValue
    @Column(name = "song_instrument_id")
    private int id;

    @ManyToOne
    @JoinColumn(name = "song_id", referencedColumnName = "song_id")
    private Song song;

    @ManyToOne
    @JoinColumn(name = "instrument_id", referencedColumnName = "instrument_id")
    private Instrument instrument;

    public SongInstrument() {}

    public SongInstrument(Song song, Instrument instrument) {
        this.song = song;
        this.instrument = instrument;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Song getSong() {
        return song;
    }

    public void setSong(Song song) {
        this.song = song;
    }

    public Instrument getInstrument() {
        return instrument;
    }

    public void setInstrument(Instrument instrument) {
        this.instrument = instrument;
    }
}

File: Frontend/src/app/Class/SongInstrument.ts
import { Song } from "./Song";
import { Instrument } from "./Instrument";

export class SongInstrument {
  id: number;
  private song: Song;
  private instrument: Instrument;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.song = (obj && obj.song) || null;
    this.instrument = (obj && obj.instrument) || null;
  }
  getId() {
    return this.id;
  }
  getSong() {
    return this.song;
  }
  getInstrument() {
    return this.instrument;
  }
  setInstrument(instrument: Instrument) {
    this.instrument = instrument;
  }
  setSong(song: Song) {
    this.song = song;
  }
}
